Userland/Libraries/LibWeb/DOM/Node.cpp

```diff
@@ -1587,4 +1590,235 @@ void Node::build_accessibility_tree(AccessibilityTreeNode& parent) const
+                current_node = node;
+                // b. Compute the text alternative of the current node beginning with step 2. Set the result to that text alternative.
+                auto result = TRY(node->name_or_description(target, document, visited_nodes));
+                // c. Append the result, with a space, to the accumulated text.
+                Node::append_with_space(total_accumulated_text, result);
+            }
+            // iii. Return the accumulated text.
+            return total_accumulated_text.to_string();
+        }
+        // C. Otherwise, if computing a name, and if the current node has an aria-label attribute whose value is not the empty string, nor, when trimmed of white space, is not the empty string:
+        if (target == NameOrDescription::Name && !element->aria_label().is_empty() && !element->aria_label().trim_whitespace().is_empty()) {
+            // TODO: If traversal of the current node is due to recursion and the current node is an embedded control as defined in step 2E, ignore aria-label and skip to rule 2E.
+            // Otherwise, return the value of aria-label.
+            return String::from_deprecated_string(element->aria_label());
+        }
+        // TODO: D. Otherwise, if the current node's native markup provides an attribute (e.g. title) or element (e.g. HTML label) that defines a text alternative,
+        //      return that alternative in the form of a flat string as defined by the host language, unless the element is marked as presentational (role="presentation" or role="none").
+
+        // TODO: E. Otherwise, if the current node is a control embedded within the label (e.g. the label element in HTML or any element directly referenced by aria-labelledby) for another widget, where the user can adjust the embedded
+        //          control's value, then include the embedded control as part of the text alternative in the following manner:
+        //   If the embedded control has role textbox, return its value.
+        //   If the embedded control has role menu button, return the text alternative of the button.
+        //   If the embedded control has role combobox or listbox, return the text alternative of the chosen option.
+        //   If the embedded control has role range (e.g., a spinbutton or slider):
+        //   If the aria-valuetext property is present, return its value,
+        //   Otherwise, if the aria-valuenow property is present, return its value,
+        //   Otherwise, use the value as specified by a host language attribute.
+
+        // F. Otherwise, if the current node's role allows name from content, or if the current node is referenced by aria-labelledby, aria-describedby, or is a native host language text alternative element (e.g. label in HTML), or is a descendant of a native host language text alternative element:
+        auto role = element->role_or_default();
+        if (role.has_value() && Web::ARIA::allows_name_from_content(role.value())) {
```

Comment:
```suggestion
        if (role.has_value() && ARIA::allows_name_from_content(role.value())) {
```
We're already in the `Web::` namespace, so you don't have to say that :)